### PR TITLE
feat(FilterConfig): Add support for grouped filters

### PR DESCRIPTION
**Pull request by bastilian**

### CODE REVIEW

File: packages/inventory-compliance/src/Utilities/ChipBuilder.js [hunk 1]
```diff
@@ -23,6 +23,24 @@ class ChipBuilder {
         chips: [{ name: this.filterConfig.labelForValue(currentValue, category) }]
     } : null)
 
+    groupFilterChip = (category, currentValue, categoryConfig) => {
+        const items = Object.keys(currentValue || {}).map((groupValue) => {
+            const group = categoryConfig.items.filter((item) => (
+                String(item.value) === String(groupValue)
+            ))[0];
```
Comment (akofink) [inline packages/inventory-compliance/src/Utilities/ChipBuilder.js:30]:
```suggestion
            const group = categoryConfig.items.find((item) => (
                String(item.value) === String(groupValue)
            ));
```

File: packages/inventory-compliance/src/Utilities/ChipBuilder.js [hunk 2]
```diff
@@ -23,6 +23,24 @@ class ChipBuilder {
...
+
+            const groupItems = Object.keys(currentValue[groupValue] || {}).map((itemValue) => (
+                group?.items.filter((item) => (String(item.value) === String(itemValue)))[0]
```
Comment (akofink) [inline packages/inventory-compliance/src/Utilities/ChipBuilder.js:33]:
```suggestion
                group?.items.find((item) => (String(item.value) === String(itemValue)))
```

**Review by akofink — COMMENTED** [review]
Seems ok, but still testing with https://github.com/RedHatInsights/compliance-frontend/pull/1131 as it evolves. Just two (very) minor comments so far.

File: packages/inventory-compliance/src/Utilities/FilterConfigBuilder.js [hunk 3]
```diff
@@ -125,7 +148,11 @@ class FilterConfigBuilder {
     )
 
     getItemByLabelOrValue = (query, category) => {
-        const items = this.getCategoryForLabel(category).items;
+        const categoryConfig = this.getCategoryForLabel(category);
+        const items = categoryConfig.type !== conditionalFilterType.group ?
+            categoryConfig.items : categoryConfig.items.flatMap((item) => (
+                item.items.map((subItem) => ({ ...subItem, osMajorVersion: item.value }))
```
Comment (akofink) [inline packages/inventory-compliance/src/Utilities/FilterConfigBuilder.js:154]: Why does this say `osMajorVersion`?

**Review by akofink — CHANGES_REQUESTED** [review]
Seems to work well with https://github.com/RedHatInsights/compliance-frontend/pull/1131. The code is not entirely straightforward to understand and seems to hardcode OS in a few places ?

File: packages/inventory-compliance/src/Utilities/FilterConfigBuilder.js (hunk 3 above)
Comment (bastilian) [reply packages/inventory-compliance/src/Utilities/FilterConfigBuilder.js:154]: Oh, i'll change that to `parentValue`. I was switching compliance and FEC too much. :D

**Review by akofink — APPROVED** [review]
Awesome, thanks! Still would appreciate some tests for this new functionality.

**Comment by bastilian** [comment]
@akofink Just added a group filter to the test config, this should cover the basics for now.

**Comment by akofink** [comment]
Thanks! :heavy_check_mark:

**Pull request merged**